### PR TITLE
#324: Created Problem Solution Service

### PR DESCRIPTION
**Pull request by afijal**

### CODE REVIEW

File: src/test/java/org/benetech/mathshare/service/ProblemSolutionServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void shouldGetProblemSolution() {
+        given(this.solutionRevisionRepository.findOneByShareCode(SolutionRevisionUtils.DEFAULT_SHARE_CODE))
+                .willReturn(SolutionRevisionUtils.createValidInstance());
+        SolutionRevision solutionRevisionFromDB =  problemSolutionService.getSolutionRevisionByShareUrl(SolutionRevisionUtils.DEFAULT_SHARE_CODE);
```
Comment (pawelcieszko) [inline src/test/java/org/benetech/mathshare/service/ProblemSolutionServiceTest.java]: remove undesired whitespace

File: src/main/java/org/benetech/mathshare/model/entity/ProblemSolution.java [hunk 2]
```diff
@@ -26,7 +25,6 @@
     @ManyToOne(cascade = CascadeType.PERSIST)
     private Problem problem;
 
-    @Lob
```
Comment (pawelcieszko) [inline src/main/java/org/benetech/mathshare/model/entity/ProblemSolution.java]: Why is it removed? Lob allows to use very long strings

Comment (afijal) [reply src/main/java/org/benetech/mathshare/model/entity/ProblemSolution.java]: It won't be needed there

File: src/test/java/org/benetech/mathshare/service/ProblemSolutionServiceTest.java (hunk 1 above)
Comment (afijal) [reply src/test/java/org/benetech/mathshare/service/ProblemSolutionServiceTest.java]: I've corrected the issue

**Pull request merged**